data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/HibernateJpaOperations.java

```diff
@@ -630,6 +631,20 @@ public <T> List<T> findAll(CriteriaQuery<T> query) {
         return executeRead(session -> session.createQuery(query).getResultList());
     }
 
+    @Override
+    public <T> List<T> findAll(CriteriaQuery<T> query, int offset, int limit) {
+        return executeRead(session -> {
+            Query<T> sessionQuery = session.createQuery(query);
+            if (offset != -1) {
+                sessionQuery = sessionQuery.setFetchSize(offset);
```

Comment:
this is wrong, it should be setFirstResult(offset). As it is, the offset is not considered at all. @dstepanov 

Would you please write a fix? Or I can do it if you are busy

Reply:
You can do, please add a test too and target 4.6.x branch. Thanks

Reply:
@dstepanov how can I target a specific branch on the fork?